tox.ini

```diff
@@ -6,25 +6,25 @@ skip_missing_interpreters = true
 
 [testenv:pylint]
 deps =
-    -r {toxinidir}/requirements_test_min.txt
-    pre-commit==2.10.1
+    -e {toxinidir}[test_minimal]
```

Comment:
IMO `tox` should always use the pinned requirements files. As explained earlier, `extras_require` should not pin exact versions.

Reply:
I'm basing this practice upon this stackoverflow answer : https://stackoverflow.com/a/28842733/2519059

I think extra_require are not always expected to be installable in alongside other packages. Especially if the extra require is called dev, test or doc.

Reply:
I believe this is outdated, as it's from 2016. Best practices for python packaging are constantly changing if existing at all. They also strongly depend on who you ask. Just take a look at the official packaging guide, it mentions `requirements` files: https://packaging.python.org/discussions/install-requires-vs-requirements/#requirements-files
Even that is probably outdated by now and AFAIK the current standard would be to use `requirements.in` files with `pip-compile`.

In conclusion, I think the current system works really well. Not sure it's really worth changing, just to be able to do `pip install -e .[tests]` when `pip install -r requirements_tests.txt` works as well.

Reply:
To be fair this is only temporary as if we move to ``pyproject.toml``and poetry which is the next step in my opinion, the dev requirement would be under ``[tool.poetry.dev-dependencies]`` (https://python-poetry.org/docs/pyproject/#dependencies-and-dev-dependencies).